addons/common/functions/fnc_runQTE.sqf

```diff
@@ -0,0 +1,103 @@
+#include "..\script_component.hpp"
+#include "\a3\ui_f\hpp\definedikcodes.inc"
+/*
+ * Author: john681611
+ * Runs a Quick time Event.
+ *
+ * Arguments:
+ * 0: Object QTE is attached to <OBJECT>
+ * 1: remoteExec command to execute <STRING>
+ * 2: remoteExec command parameters <ARRAY>
+ * 3: remoteExec command target <STRING>
+ * 4: QTE Code <ARRAY>
+ * 5: Max interaction distance from attached object <NUMBER> (default: 10) 
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [car, "hint", "Yay", 1, ["↑", "↓", "→", "←"], 15] call ace_common_fnc_runQTE
+ *
+ * Public: Yes
+ */
+
+params ["_veh", "_command", "_command_params", "_command_exec_code", "_keyCode", ["_max_distance", 10]];
+
+GVAR(QTEVeh) = _veh;
+GVAR(QTECommand) = _command;
+GVAR(QTECommandParams) = _command_params;
+GVAR(QTECommandExecCode) = _command_exec_code;
+GVAR(QTEMaxDistance) = _max_distance;
+GVAR(QTECode) = _keyCode;
+GVAR(QTEHistory) = [];
+
+hint format ["%1: \n %2", localize LSTRING(QTEInput), [GVAR(QTECode)] call FUNC(getFormattedQTECode)];
+
+findDisplay 46 displayAddEventHandler ["KeyUp", {
+    params ["_displayOrControl", "_key", "_shift", "_ctrl", "_alt"];
+    private _override = false;
+    private _inputKeys = [DIK_UP, DIK_DOWN, DIK_LEFT, DIK_RIGHT];
+    if (ACE_player distance GVAR(QTEVeh) > GVAR(QTEMaxDistance)) exitWith {
+        hint (localize LSTRING(QTECancelled));
+        findDisplay 46 displayRemoveEventHandler ["KeyUp", _thisEventHandler];
+        _override;
+    };
+
+    switch (GVAR(QTEInputKeys)) do {
+        case "WASD": {
+            _inputKeys = [DIK_W, DIK_S, DIK_A, DIK_D];
+        };
+        case "IJKL": {
+            _inputKeys = [DIK_I, DIK_K, DIK_J, DIK_L];
+        };
+        case "NUMPAD": {
+            _inputKeys = [DIK_NUMPAD8, DIK_NUMPAD2, DIK_NUMPAD4, DIK_NUMPAD6];
+        };
+        default {
+            _inputKeys = [DIK_UP, DIK_DOWN, DIK_LEFT, DIK_RIGHT];
+        };
+    };
```

Comment:
Imo, it's a waste of performance - only use 1 system of input.
Given as a majority of players use WASD to move, I'd stick with that, but then the arrow characters in the code don't make immediate sense.
We need more opinions in this though.

Reply:
Just based on the players in my small squad we have:
an Arrow user and a IJKL user just for movement.

it's also worth noting that with Hell Divers 2 (where the inspiration came from) it feels much more natural to have the input and movement sets separated (WASD for movement & Arrows for QTE input)